### DISCUSSION TITLE
How to use upsert without an id?

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#9763] [Q&A] [answered]
I have this mutation:
```ts
  addMovieRating: async (_, args, {req, res}) => {
    const movieRating = await prisma.movieRating.upsert({
      where: {
        id: args.id,
      },
      update: {
        value: args.value,
      },
      create: {
        value: args.value,
        userId: args.userId,
        movieId: args.movieId,
        user: {
          connect: {id: args.userId},
        },
        movie: {
          connect: {id: args.movieId},
        },
      },
    });

    return movieRating;
  },
  ```

My movieRating model:
```prisma
model MovieRating {
  id              Int           @id @default(autoincrement())
  value           Int           @default(0)
  user            User[]        @relation(fields: [userId], references: [id])
  userId          Int
  movie           Movie[]       @relation(fields: [movieId], references: [id])
  movieId         Int
}
```
A movie rating has a many to many relation to Users and Movies:
![image](https://user-images.githubusercontent.com/779921/137176274-11c11047-f059-46ff-bf47-9d560be50452.png)

A rating can be created or the value of a rating can be updated. In my mutation I have a ` where`  with an id value. But the id value only exists if the rating was already created. So how would I create the first movieRating record?

**Comment by alucardu** [comment, marked as answer]
This seems to work:
```
where: {
  id: args.id || 0,
},
```
If an id is provided it will set it, otherwise it will use id 0 which does not exist so it will create a new record.

**Reply by ryands17** [reply]
Yes that is the way to do it as of now. It would be great if you could add a 👍 to this request so that we know the priority for this: https://github.com/prisma/prisma/issues/5233

**Reply by rubenvar** [reply]
In case this helps anyone:

I faced this same issue and my first idea was to try a similar solution. But I am using MongoDB, with ids of type ObjectId. So using a `0` or a random `string` doesn't work.

Finally, I solved it by adding the `bson` package and generating a valid ObjectId:

```ts
import { ObjectId } from 'bson';

/* ... */

where: { id: args.id || new ObjectId().toString() }
```

I guess that, as long as the generated ObjectId is not found in the db, this should work. I understand it's not ideal.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
